src/validators.js

```diff
@@ -268,20 +263,29 @@ function validateQualifiedMinCountProperty ($context, $this, $path, $qualifiedVa
 
 function validateQualifiedHelper ($context, $this, $path, $qualifiedValueShape, $qualifiedValueShapesDisjoint, $currentShape) {
   const siblingShapes = new NodeSet()
+
   if ($context.factory.term('true').equals($qualifiedValueShapesDisjoint)) {
-    $context.$shapes.query()
-      .match('?parentShape', 'sh:property', $currentShape)
-      .match('?parentShape', 'sh:property', '?sibling')
-      .match('?sibling', 'sh:qualifiedValueShape', '?siblingShape')
-      .filter(RDFQuery.exprNotEquals('?siblingShape', $qualifiedValueShape))
-      .addAllNodes('?siblingShape', siblingShapes)
+    // TODO: I'm really not sure of this name
```

Comment:
`siblingShapes` + `qualifiedValueShape` -> `qualifiedSiblingShapes ` makes sense to me